### ISSUE TITLE
Parse exception when add new field

### ISSUE THREAD

**Issue by esvalukhin** [jshiell/checkstyle-idea#185]
I am getting following exception when add new field to existing class after updating plugin to version 4.19.0. On 4.18.0 version it works perfect. Do I notify checkstyle team about error?

The inspection could not be executed.
```
org.infernus.idea.checkstyle.exception.CheckStylePluginException: The inspection could not be executed.
	at org.infernus.idea.checkstyle.CheckStylePlugin.processError(CheckStylePlugin.java:140)
	at org.infernus.idea.checkstyle.CheckStyleInspection.checkFile(CheckStyleInspection.java:92)
	at com.intellij.codeInspection.LocalInspectionTool$1.visitFile(LocalInspectionTool.java:151)
	at com.intellij.psi.impl.source.PsiJavaFileBaseImpl.accept(PsiJavaFileBaseImpl.java:427)
	at com.intellij.codeInspection.InspectionEngine.acceptElements(InspectionEngine.java:80)
	at com.intellij.codeInspection.InspectionEngine.createVisitorAndAcceptElements(InspectionEngine.java:69)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.a(LocalInspectionsPass.java:285)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.access$200(LocalInspectionsPass.java:75)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass$2.process(LocalInspectionsPass.java:254)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass$2.process(LocalInspectionsPass.java:251)
	at com.intellij.concurrency.ApplierCompleter.a(ApplierCompleter.java:122)
	at com.intellij.concurrency.ApplierCompleter.access$000(ApplierCompleter.java:44)
	at com.intellij.concurrency.ApplierCompleter$1.run(ApplierCompleter.java:85)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1098)
	at com.intellij.concurrency.ApplierCompleter$2.run(ApplierCompleter.java:94)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:452)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:449)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:402)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
	at com.intellij.concurrency.ApplierCompleter.a(ApplierCompleter.java:106)
	at com.intellij.concurrency.ApplierCompleter.compute(ApplierCompleter.java:82)
	at jsr166e.CountedCompleter.exec(CountedCompleter.java:684)
	at jsr166e.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at jsr166e.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:858)
	at jsr166e.ForkJoinPool.scan(ForkJoinPool.java:1687)
	at jsr166e.ForkJoinPool.runWorker(ForkJoinPool.java:1642)
	at jsr166e.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:109)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: MismatchedTokenException occurred during the analysis of file /tmp/csi-de3dead4-97bd-4c55-8f84-cbd41524b980/src/main/java/kz/inessoft/egov/pep/jbpm/config/AbstractWorkItemHandlerProducer.java.
	at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:222)
	at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:79)
	at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:265)
	at org.infernus.idea.checkstyle.checker.CheckStyleChecker.processAndAudit(CheckStyleChecker.java:64)
	at org.infernus.idea.checkstyle.checker.CheckStyleChecker.scan(CheckStyleChecker.java:40)
	at org.infernus.idea.checkstyle.CheckStyleInspection.lambda$checkFile$14(CheckStyleInspection.java:73)
	at org.infernus.idea.checkstyle.CheckStyleInspection$$Lambda$44/1168851188.apply(Unknown Source)
	at java.util.Optional.map(Optional.java:215)
	at org.infernus.idea.checkstyle.CheckStyleInspection.checkFile(CheckStyleInspection.java:73)
	... 25 more
Caused by: /tmp/csi-de3dead4-97bd-4c55-8f84-cbd41524b980/src/main/java/kz/inessoft/egov/pep/jbpm/config/AbstractWorkItemHandlerProducer.java:23:5: expecting EOF, found '}'
	at antlr.Parser.match(Parser.java:211)
	at com.puppycrawl.tools.checkstyle.grammars.GeneratedJavaRecognizer.compilationUnit(GeneratedJavaRecognizer.java:210)
	at com.puppycrawl.tools.checkstyle.TreeWalker.parse(TreeWalker.java:470)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:205)
	... 33 more
```

**Comment by jshiell** [comment]
The Checkstyle teams official word is that Checkstyle only works with parseable files. So I fear I'll have to add more workarounds to catch and suppress these errors. I'll try have a look at this this weekend.

Thanks for the report!

**Comment by jshiell** [comment]
Given I've now a definitive answer from the Checkstyle devs - that behaviour is undefined when the file is unparseable (checkstyle:127) - I've changed the error handling a bit. We should now ignore everything but CheckstyleExceptions, and we should filter out the CheckstyleExceptions that are Antlr parse problems (based on the TreeWalker code, anyway). So I hope this will prove a little more robust and limit these sorts of problems in future.

Released as 4.19.1 - please shout if you have any problems with it, as I'm sure my testing has covered only a fraction of the edge cases here.
